src/series.vala

```diff
@@ -2,23 +2,27 @@ using Gee;
 namespace LiveChart { 
 
     public class Series : Object {
-
+        private Gee.Map<Serie, ulong> signals = new Gee.HashMap<Serie, ulong>();
         private Gee.ArrayList<Serie> series = new Gee.ArrayList<Serie>();
-        private Chart chart;
+        private weak Chart chart;
 
         public Series(Chart chart) {
             this.chart = chart;
         }
 
         public Serie register(Serie serie) {
+            if(signals.has_key(serie)){
```

Comment:
IMHO, for clarity, I would just use something like `if(series.contains(serie))` tocheck if the serie has already been registered. IMHO, using signals to do that check is a bit confusing :)

Reply:
I see.
（I just disliked to use List.contains=performing linear searching. However, in this case, no problem to use it because it is not in the rendering proc.）